native/src/main/java/io/ballerina/stdlib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,350 @@
+            for (int i = 0; i < headers.length; i++) {
+                headers[i] = array.get(i).toString();
+            }
+        }
+
+        return headers;
+    }
+
+    public static Optional<Type> getMutableType(IntersectionType intersectionType) {
+        for (Type constituentType : intersectionType.getConstituentTypes()) {
+            if (constituentType.getTag() == TypeTags.READONLY_TAG) {
+                continue;
+            }
+            return Optional.of(constituentType);
+        }
+        return Optional.empty();
+    }
+
+    public static Object convertToBasicType(Object csv, Type targetType, CsvConfig config) {
+        if (csv == null) {
+            csv = config.nilValue;
+        }
+        return ValueUtils.convert(csv, targetType);
+    }
+
+    public static void checkRequiredFieldsAndLogError(Map<String, Field> filedHierarchy, boolean absentAsNilableType) {
+        filedHierarchy.values().forEach(field -> {
+            if (absentAsNilableType && field.getFieldType().isNilable()) {
+                return;
+            }
+            if (SymbolFlags.isFlagOn(field.getFlags(), SymbolFlags.REQUIRED)) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_FIELD_IN_CSV, field.getFieldName());
+            }
+        });
+    }
+
+    public static boolean isHeaderFieldsEmpty(Map<String, Field> currentField) {
+        for (Field field: currentField.values()) {
+            if (SymbolFlags.isFlagOn(field.getFlags(), SymbolFlags.REQUIRED)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean checkTypeCompatibility(Type constraintType, Object csv, boolean stringConversion) {
+        int tag = constraintType.getTag();
+        if ((csv instanceof BString && (stringConversion || tag == TypeTags.STRING_TAG
+                || tag == TypeTags.CHAR_STRING_TAG || isJsonOrAnyDataOrAny(tag)))
+                || (csv instanceof Long && (tag == TypeTags.INT_TAG
+                || tag == TypeTags.FLOAT_TAG || tag == TypeTags.DECIMAL_TAG || tag == TypeTags.BYTE_TAG
+                || tag == TypeTags.SIGNED8_INT_TAG || tag == TypeTags.SIGNED16_INT_TAG
+                || tag == TypeTags.SIGNED32_INT_TAG || tag == TypeTags.UNSIGNED8_INT_TAG
+                || tag == TypeTags.UNSIGNED16_INT_TAG || tag == TypeTags.UNSIGNED32_INT_TAG
+                || isJsonOrAnyDataOrAny(tag)))
+                || (csv instanceof BDecimal && ((tag == TypeTags.DECIMAL_TAG
+                || tag == TypeTags.FLOAT_TAG || tag == TypeTags.INT_TAG) || isJsonOrAnyDataOrAny(tag)))
+                || (csv instanceof Double && ((tag == TypeTags.FLOAT_TAG
+                || tag == TypeTags.DECIMAL_TAG || tag == TypeTags.INT_TAG) || isJsonOrAnyDataOrAny(tag)))
+                || (Boolean.class.isInstance(csv) && (tag == TypeTags.BOOLEAN_TAG || isJsonOrAnyDataOrAny(tag)))
+                || (csv == null && (tag == TypeTags.NULL_TAG || isJsonOrAnyDataOrAny(tag)))) {
```

Comment:
If possible lets factor out this condition to a separate method and give it a meaningful name

Reply:
The purposed oof this function is that.
Anyway, I simplified the method by directly returning that.